Gems/Atom/RPI/Code/Include/Atom/RPI.Public/GpuQuery/GpuPassProfiler.h

```diff
@@ -0,0 +1,132 @@
+                Name m_path;
+
+                RPI::TimestampResult m_timestampResult;
+                uint64_t m_interpolatedTimestampInNanoseconds = 0u;
+
+                //! Convert the PipelineStatistics result to an array for easier access.
+                PipelineStatisticsArray m_pipelineStatistics;
+
+                //! Used as a double linked structure to reference the parent <-> child relationship.
+                PassEntry* m_parent = nullptr;
+                AZStd::vector<PassEntry*> m_children;
+
+                //! Mirrors the enabled queries state of the pass.
+                bool m_timestampEnabled = false;
+                bool m_pipelineStatisticsEnabled = false;
+
+                //! Mirrors the enabled/disabled state of the pass.
+                bool m_enabled = false;
+
+                //! Dirty flag to determine if this entry is linked to an parent entry.
+                bool m_linked = false;
+
+                //! Cache if the pass is a parent.
+                bool m_isParent = false;
+            };
+
+            GpuPassProfiler() = default;
+            ~GpuPassProfiler() = default;
+
+            GpuPassProfiler(GpuPassProfiler&& other) = delete;
+            GpuPassProfiler(const GpuPassProfiler& other) = delete;
+            GpuPassProfiler& operator=(const GpuPassProfiler& other) = delete;
+
+            void SetRenderPipelineGpuTimeMeasurementEnabled(bool enabled) { m_measureRenderPipelineGpuTime = enabled; }
+            bool IsRenderPipelineGpuTimeMeasurementEnabled() { return m_measureRenderPipelineGpuTime; }
+
+            // Measures the total time spent by the Render Pipeline inside the GPU.
+            // It measures the time by using the functions CreatePassEntriesDatabase(), SortPassEntriesByTimestamps() & CalculateTotalGpuPassTime().
+            // The above functions are public and exist in split form to faciliate integration in tools like the ImGuiGpuProfiler, etc.
+            // if @m_measureRenderPipelineGpuTime is false this function returns 0.
+            uint64_t MeasureRenderPipelineGpuTimeInNanoseconds(RHI::Ptr<RPI::ParentPass> rootPass);
+
+            //////////////////////////////////////////////////////////////////
+            // Support Functions START
+            // The following functions provide all the details when measuring
+            // the render pipeline performance.
+            
+            // Returns the pass entry database where the key is PathName -> value is PassEntry
+            AZStd::unordered_map<Name, PassEntry> CreatePassEntriesDatabase(RHI::Ptr<RPI::ParentPass> rootPass);
+
+            // Measures the time spent in the GPU for each pass.
+            // Returns the total amount of nanoseconds spent in the GPU for the Root Parent Pass.
+            AZStd::vector<GpuPassProfiler::PassEntry*> SortPassEntriesByTimestamps(AZStd::unordered_map<Name, PassEntry>& timestampEntryDatabase);
```

Comment:
The comment doesn't match the function name.

Reply:
Fixed.